source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -218,6 +249,10 @@ InstanceImpl::ThreadLocalPool::makeRequestToHost(const std::string& host_address
     client->redis_client_ =
         parent_.client_factory_.create(it->second, dispatcher_, parent_.config_);
     client->redis_client_->addConnectionCallbacks(*client);
+    if (!auth_password_.empty()) {
```

Comment:
Nit: Is it worth breaking the whole block `ThreadLocalActiveClientPtr...` into a function `authIfNoClient`?

Reply:
I'll move all of this logic into a threadLocalActiveClient() method for InstanceImpl::ThreadLocalPool ...